src/cli/service/serve.rs

```diff
@@ -0,0 +1,370 @@
+    ///
+    /// Or using HaProxy protocol.
+    pub fn forward(self, kind: ForwardKind) -> Self {
+        self.maybe_forward(Some(kind))
+    }
+
+    /// enable support for one of the following "forward" headers or protocols
+    ///
+    /// Same as [`Self::forward`] but without consuming `self`.
+    pub fn set_forward(&mut self, kind: ForwardKind) -> &mut Self {
+        self.forward = Some(kind);
+        self
+    }
+
+    /// maybe enable support for one of the following "forward" headers or protocols.
+    ///
+    /// See [`Self::forward`] for more information.
+    pub fn maybe_forward(mut self, maybe_kind: Option<ForwardKind>) -> Self {
+        self.forward = maybe_kind;
+        self
+    }
+
+    #[cfg(any(feature = "rustls", feature = "boring"))]
+    /// define a tls server cert config to be used for tls terminaton
+    /// by the serve service.
+    pub fn tls_server_config(mut self, cfg: TlsConfig) -> Self {
+        self.tls_server_config = Some(cfg);
+        self
+    }
+
+    #[cfg(any(feature = "rustls", feature = "boring"))]
+    /// define a tls server cert config to be used for tls terminaton
+    /// by the serve service.
+    pub fn set_tls_server_config(&mut self, cfg: TlsConfig) -> &mut Self {
+        self.tls_server_config = Some(cfg);
+        self
+    }
+
+    #[cfg(any(feature = "rustls", feature = "boring"))]
+    /// define a tls server cert config to be used for tls terminaton
+    /// by the serve service.
+    pub fn maybe_tls_server_config(mut self, cfg: Option<TlsConfig>) -> Self {
+        self.tls_server_config = cfg;
+        self
+    }
+
+    /// set the http version to use for the http server (auto by default)
+    pub fn http_version(mut self, version: Version) -> Self {
+        self.http_version = Some(version);
+        self
+    }
+
+    /// maybe set the http version to use for the http server (auto by default)
+    pub fn maybe_http_version(mut self, version: Option<Version>) -> Self {
+        self.http_version = version;
+        self
+    }
+
+    /// set the http version to use for the http server (auto by default)
+    pub fn set_http_version(&mut self, version: Version) -> &mut Self {
+        self.http_version = Some(version);
+        self
+    }
+
+    /// add a custom http layer which will be applied to the existing http layers
+    pub fn http_layer<H2>(self, layer: H2) -> ServeServiceBuilder<(H, H2)> {
+        ServeServiceBuilder {
+            concurrent_limit: self.concurrent_limit,
+            body_limit: self.body_limit,
+            timeout: self.timeout,
+            forward: self.forward,
+
+            #[cfg(any(feature = "rustls", feature = "boring"))]
+            tls_server_config: self.tls_server_config,
+
+            http_version: self.http_version,
+
+            http_service_builder: (self.http_service_builder, layer),
+
+            content_path: self.content_path,
+        }
+    }
+
+    pub fn content_path(mut self, path: PathBuf) -> Self {
```

Comment:
You can use `impl Into<PathBuf>` to also allow other stuff, makes it easier, you can do this also for `set_content_path`. However `maybe_content_path` better is left like it is as otherwise `None` will not know what type to use.